controller_manager:
  ros__parameters:
    update_rate: null

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    hand_guide_position_controller:
      type: rvim_position_controllers/HandGuidePositionController

    collaborative_twist_position_controller:
      type: rvim_position_controllers/CollaborativeTwistPositionController

hand_guide_position_controller:
  ros__parameters:
    joints:
      - lbr_joint_0
      - lbr_joint_1
      - lbr_joint_2
      - lbr_joint_3
      - lbr_joint_4
      - lbr_joint_5
      - lbr_joint_6
    chain_root: lbr_link_0
    chain_tip: lbr_link_ee
    command_interface: position
    state_interfaces: 
      - position
      - external_torque

collaborative_twist_position_controller:
  ros__parameters:
    joints:
      - lbr_joint_0
      - lbr_joint_1
      - lbr_joint_2
      - lbr_joint_3
      - lbr_joint_4
      - lbr_joint_5
      - lbr_joint_6
    force_constraint_relaxation: 0.  # N
    torque_constraint_relaxation: 0.  # Nm
    force_threshold: 2.0  # N
    torque_threshold: 0.5  # Nm
    dq_lim: 0.02
    mu: 0.01
    n_classes: 4
    pos_stiffness_ee: 3000.
    ori_stiffness_ee: 300.
    pos_stiffness_cam: 300.
    ori_stiffness_cam: 50.
    base_link: lbr_link_0
    hand_guide_link: lbr_link_ee
    camera_link: storz_straight_exoscope_hsi_link_cm_optical
    command_interface: position
    state_interfaces:
      - position
      - external_torque
